addons/purchase_stock/__init__.py

```diff
@@ -12,5 +12,10 @@ def _create_buy_rules(env):
     necessary if the purchase_stock module is installed after some warehouses
     were already created.
     """
-    warehouse_ids = env['stock.warehouse'].search([('buy_pull_id', '=', False)])
+    all_warehouses = env['stock.warehouse'].search([])
+    buy_route = env['stock.route'].search([('name', '=', 'Buy')])
```

Comment:
mm, there's a good chance this may not work as you expect. E.g. `purchase` is installed when the db is not in English ('Buy' will be translated in that case) or if the user creates a custom route with that name

Reply:
Makes sense. I had to change it to search for the route whose all rules' actions are `'buy'`. I feel it's a bit slower but not sure if there's a better way 😕.

Reply:
Yeah, I'm not sure either. I've never really worked directly with the `'buy'` route much myself :shrug: Although it contradicts what arm said earlier, it might be safer to directly reference `route_warehouse0_buy` and then fallback to searching for a buy if the record isn't found :thinking: